lib/DxilValidation/DxilValidation.cpp

```diff
@@ -970,6 +970,226 @@ static void ValidateImmOperandForMathDxilOp(CallInst *CI, DXIL::OpCode Opcode,
   }
 }
 
+static bool CheckFromRegisterInterpretations(uint32_t Input) {
+  DXIL::ComponentType ValidSet[] = {
+      DXIL::ComponentType::I16,         DXIL::ComponentType::U16,
+      DXIL::ComponentType::I32,         DXIL::ComponentType::U32,
+      DXIL::ComponentType::F16,         DXIL::ComponentType::F32,
+      DXIL::ComponentType::PackedS8x32, DXIL::ComponentType::PackedU8x32,
+      DXIL::ComponentType::U8,          DXIL::ComponentType::I8,
+      DXIL::ComponentType::F8_E4M3,     DXIL::ComponentType::F8_E5M2};
+
+  for (auto Val : ValidSet) {
+    if (Val == static_cast<DXIL::ComponentType>(Input))
+      return true;
+  }
+  return false;
+}
+
+static bool CheckInMemoryInterpretations(uint32_t Input) {
+  DXIL::ComponentType ValidSet[] = {
+      DXIL::ComponentType::I16,     DXIL::ComponentType::U16,
+      DXIL::ComponentType::I32,     DXIL::ComponentType::U32,
+      DXIL::ComponentType::F16,     DXIL::ComponentType::F32,
+      DXIL::ComponentType::U8,      DXIL::ComponentType::I8,
+      DXIL::ComponentType::F8_E4M3, DXIL::ComponentType::F8_E5M2};
+
+  for (auto Val : ValidSet) {
+    if (Val == static_cast<DXIL::ComponentType>(Input))
+      return true;
+  }
+  return false;
+}
+
+static bool CheckMatrixLayout(unsigned Input) {
+  return (Input <=
+          static_cast<unsigned>(DXIL::DXILMatrixLayout::OuterProductOptimal));
+}
+
+static bool CheckTransposeForMatrixLayout(DXIL::DXILMatrixLayout Layout,
+                                          bool Transposed) {
+  switch (Layout) {
+  case DXIL::DXILMatrixLayout::RowMajor:
+  case DXIL::DXILMatrixLayout::ColumnMajor:
+    return !Transposed;
+
+  default:
+    return true;
+  }
+}
+
+static bool CheckUnsignedFlag(Type *VecTy, bool IsUnsigned) {
+  Type *ElemTy = VecTy->getScalarType();
+  if (ElemTy->isFloatingPointTy())
+    return !IsUnsigned;
+
+  return true;
+}
+
+static Value *getMatVecOpIsOutputUnsigned(CallInst *CI, DXIL::OpCode OpCode) {
```

Comment:
nit: All the other functions you've added are PascalCase, this one is camelCase. The coding conventions are camelCase, but if the rest of the file is all PascalCase we should match.
(see: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly)
```suggestion
static Value *GetMatVecOpIsOutputUnsigned(CallInst *CI, DXIL::OpCode OpCode) {
```